plugins/polio/api/vaccines/repository.py

```diff
@@ -0,0 +1,260 @@
+from datetime import datetime
+
+from django.db.models import Q, Min, Max, OuterRef, Subquery
+from django.utils.decorators import method_decorator
+from django.views.decorators.cache import cache_page
+from drf_yasg import openapi
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import permissions, serializers, filters
+from rest_framework.filters import OrderingFilter, SearchFilter
+from rest_framework.mixins import ListModelMixin
+from rest_framework.viewsets import GenericViewSet
+from datetime import timedelta
+
+from iaso.api.common import Paginator
+from plugins.polio.models import (
+    Campaign,
+    CampaignType,
+    DestructionReport,
+    IncidentReport,
+    OutgoingStockMovement,
+    Round,
+    VaccinePreAlert,
+    VaccineRequestForm,
+    VaccineRequestFormType,
+)
+
+
+class VaccineRepositorySerializer(serializers.Serializer):
+    country_name = serializers.CharField(source="campaign.country.name")
+    campaign_obr_name = serializers.CharField(source="campaign.obr_name")
+    round_id = serializers.IntegerField(source="id")
+    round_number = serializers.IntegerField(source="number")
+    start_date = serializers.DateField(source="started_at")
+    end_date = serializers.DateField(source="ended_at")
+
+    vrf_data = serializers.SerializerMethodField()
+    pre_alert_data = serializers.SerializerMethodField()
+    form_a_data = serializers.SerializerMethodField()
+
+    def get_vrf_data(self, obj):
+        vrfs = VaccineRequestForm.objects.filter(campaign=obj.campaign)
+        return [
+            {
+                "date": vrf.date_vrf_reception,
+                "file": vrf.document.url if vrf.document else None,
+                "is_missing": vrf.vrf_type == VaccineRequestFormType.MISSING,
+                "is_not_required": vrf.vrf_type == VaccineRequestFormType.NOT_REQUIRED,
+            }
+            for vrf in vrfs
+        ]
+
+    def get_pre_alert_data(self, obj):
+        pre_alerts = VaccinePreAlert.objects.filter(request_form__campaign=obj.campaign)
+        return [
+            {"date": pa.date_pre_alert_reception, "file": pa.document.url if pa.document else None} for pa in pre_alerts
+        ]
+
+    def get_form_a_data(self, obj):
+        form_as = OutgoingStockMovement.objects.filter(campaign=obj.campaign, round=obj)
```

Comment:
`VaccineRepositorySerializer` is issuing one extra SQL query per `get_*` method.

This results in a N+3 for each result in the list view.

This won't scale well.

Is there an easy way to reduce the number of queries?